### DISCUSSION TITLE
가끔 요청을 보낼때 서버에서 401 Unauthorized라고 합니다.

### DISCUSSION THREAD

**Discussion by 0114kek** [wafflestudio/seminar-2021#625] [Q&A] [answered]
안녕하세요. 서버로 요청을 보냈을때 가끔 401 Unauthorized가 발생합니다.

Dashboard의 데이터를 3초마다 요청하는 부분을 구현하다가 발생했습니다.

아래 사진의 stat는 학생 전체 정보를 요청하는 것이고
check_token은 토큰 만료 여부를 확인합니다.
![image](https://user-images.githubusercontent.com/88540493/140037116-b8e63f81-84d5-41a3-8bce-bc3b84c3754c.png)

빨간색으로 표시된 오류가 가끔 발생합니다.
response가 사진과 같이 나타나지 않아 빨간 글씨를 더블클릭해서 확인해보니 아래와 같은 창이 뜹니다. 
![2](https://user-images.githubusercontent.com/88540493/140037277-8499d031-71cc-41f9-b237-42c9fcde018d.PNG)

401 Unauthorized라고 하여 제가 토큰을 잘못줬나 혹은 토큰이 만료되었나 확인해봤는데 그건 또 아니었습니다.

![3](https://user-images.githubusercontent.com/88540493/140037547-bae538eb-203d-4d86-906b-4dd04e1097ad.PNG)
위의 사진처럼 정상적으로 토큰이 들어갔고 이 이후의 요청들은 동일한 토큰을 이용해 정상인 응답을 받았습니다.
즉 정상적인 토큰을 줬는데도 401 Unauthorized라고 응답했습니다....

이 현상이 어떨 때는 1분사이에 두번씩 일어나기도 하고 어떨 때는 10분넘게 일어나지 않기도 합니다.

실제로 문제가 되는 부분은 stat이 아니라 함께 요청하는 check_token에서 오류가 발생하는 경우입니다.
이로 인해 토큰이 만료된것으로 생각하고 로그아웃 시켜버리는 현상이 가끔 나타납니다.....

**Comment by woohm402** [comment, marked as answer]
일단 check_token 은 처음에 앱이 마운트될 때만 호출하시라는 의도로 만든 api입니다. 굳이 자주 호출할 필요가 없습니다.

올려주신 걸로 봐서는 원인이 잘 파악되지 않는데.. 일단 무시하고 진행해주셔도 될 것 같습니다.

**Reply by 0114kek** [reply]
감사합니다.
그렇다면 3초마다 stat요청과 함께 굳이 check_token을 하지 않아도 상관없을까요?
즉 10분이 지나 stat 요청이 만료된 토큰과 함께가서 나타나는 에러는 무시해도 될까요?

**Reply by woohm402** [reply]
스펙상 토큰이 만료되었다고 해서 자동 로그아웃이 될 필요는 없습니다. 새로고침하면 아마 로그아웃이 되겠죠?

물론 구현해주시면 좋긴 합니다만, 그러려면 `check_token` 말고 `stat` 날려서 `401 Unauthorized` 뜨면 자동 로그아웃하는 방식이 좋을 것 같네요!
